Source/WebCore/dom/StaticRange.cpp

```diff
@@ -83,4 +83,21 @@ void StaticRange::visitNodesConcurrently(JSC::AbstractSlotVisitor& visitor) cons
     addWebCoreOpaqueRoot(visitor, end.container.get());
 }
 
+bool StaticRange::computeValidity() const
+{
+    Node& startContainer = this->startContainer();
+    Node& endContainer = this->endContainer();
+
+    // Its start is before or equal to its end.
+    if (endOffset() < startOffset() && &startContainer == &endContainer)
+        return false;
+    if (is_gt(treeOrder<ComposedTree>(startContainer, endContainer)))
+        return false;
+    // Its start and end are in the same node tree.
+    if (!connectedInSameTreeScope(&startContainer.rootNode(), &endContainer.rootNode()))
+        return false;
+    // Its start offset is between 0 and its start node’s length, inclusive.
+    // Its end offset is between 0 and its end node’s length, inclusive.
+    return endOffset() <= endContainer.length() && startOffset() <= startContainer.length();
```

Comment:
And this check before checking is_gt(treeOrder<ComposedTree>(...), which is O(n).